File: src/components/FileList/FileList.js
import React from "react";
import { makeStyles } from "@material-ui/core/styles";

import styles from "assets/jss/material-kit-react/views/mathPageSections/mathSectionStyle.js";

import file0 from "assets/docs/Задание 13. Уравнение.pdf";
import file1 from "assets/docs/Задание 15. Неравенства.pdf";
import file2 from "assets/docs/Задание 15. Показательные и логарифмические неравенства.pdf";
import file3 from "assets/docs/Задание 17. Вклады и кредиты.pdf";
import file4 from "assets/docs/Иррациональные ур. и нер.pdf";
import file5 from "assets/docs/Логарифмы.pdf";
import file6 from "assets/docs/Метод интервалов.pdf";
import file7 from "assets/docs/метод рационализации.pdf";

const useStyles = makeStyles(styles);

export default function FileList() {
  const classes = useStyles();
  return (
    <ul className={classes.uList}>
      <li>
        {" "}
        <a href={file0} target="_blank" className={classes.anchor}>
          Задание 13. Уравнение
        </a>
      </li>

      <li>
        {" "}
        <a href={file1} target="_blank" className={classes.anchor}>
          Задание 15. Неравенства
        </a>
      </li>

      <li>
        {" "}
        <a href={file2} target="_blank" className={classes.anchor}>
          Задание 15. Показательные и логарифмические неравенства
        </a>
      </li>

      <li>
        {" "}
        <a href={file3} target="_blank" className={classes.anchor}>
          Задание 17. Вклады и кредиты
        </a>
      </li>

      <li>
        {" "}
        <a href={file4} target="_blank" className={classes.anchor}>
          Иррациональные ур. и нер
        </a>
      </li>

      <li>
        {" "}
        <a href={file5} target="_blank" className={classes.anchor}>
          Логарифмы
        </a>
      </li>

      <li>
        {" "}
        <a href={file6} target="_blank" className={classes.anchor}>
          Метод интервалов
        </a>
      </li>

      <li>
        {" "}
        <a href={file7} target="_blank" className={classes.anchor}>
          метод рационализации
        </a>
      </li>
    </ul>
  );
}

File: src/components/FileList/fileList.py
import pathlib
import os
from os.path import isfile, join
from jinja2 import Template
import codecs
import regex as re
from transliterate import translit, get_available_language_codes

# dirname = "../../assets/docs/" + directory

# dirname = dirname.replace("\\", "/")
#dirname = dirname.replace(" ", "\ ")
#dirname = dirname.replace("\\", "/")
# dirname = dirname.replace("c:", "/c")
# dirname = re.sub("(\w+\s\w+)", "'\\1'", dirname)

# print(filesInDir)

listTmpl = Template("""
    {% for i in range(n) %}
    <li>
        {" "}
        <a href={ {{file[i]}} } target="_blank" rel="noopener noreferrer" className={classes.anchor}>
            {{filenames[i].replace('.pdf', '')}}
        </a>
    </li>
    {% endfor %}
""")

def genFileList(dirname):
    mypath = f'../../assets/docs/{dirname}'
    filesInDir = [f for f in reversed(os.listdir(mypath))
              if isfile(join(mypath, f))]
    # print(dirname, filesInDir)

    toImport = []
    JSfilename = 'FileList' + re.sub(' ', '', translit(dirname, reversed=True).title())
    with codecs.open(f'{JSfilename}.js', 'w', 'utf-8') as fw:
        fw.write("""import React from "react";
import { makeStyles } from "@material-ui/core/styles";

import styles from "assets/jss/material-kit-react/views/mathPageSections/mathSectionStyle.js";\n\n""")
        for i, name in enumerate(filesInDir):
            toImport.append(f'file{i}')
            fw.write(f'import file{i} from \"assets/docs/{dirname}/{name}\";\n')
        fw.write("""

const useStyles=makeStyles(styles);

export default function """ + JSfilename + """() {

const classes=useStyles();
return(
    <ul className={classes.uList} > """)
        fw.write(listTmpl.render(
            file=toImport,
            filenames=filesInDir,
            n=(len(filesInDir))
        ))
        fw.write("""</ul>
        );
    }
        """)

with codecs.open('../../views/MathPage/Sections/Mathsection.js', 'r', 'utf-8') as fr:
    data = fr.readlines()

for k, dirname in enumerate(os.listdir("../../assets/docs/")):
    # print(dirname)
    genFileList(dirname)
    trans = 'FileList' + re.sub(' ', '', translit(dirname, reversed=True).title())
    data.insert(17, f'import {trans} from \"components/FileList/{trans}.js\";\n' )

with codecs.open('../../views/MathPage/Sections/Mathsection.js', 'w', 'utf-8') as fw:
    fw.writelines(data)
